payment/service.go

```diff
@@ -750,92 +750,80 @@ func (s *Service) GetCredentialSigningKeys(ctx context.Context, merchantID strin
 	return resp, nil
 }
 
-var (
-	errInvalidIssuanceInterval = errors.New("invalid issuance interval")
-	errOrderExpired            = errors.New("order is expired")
-)
-
 // credChunkFn - given a time, calculate the next increment of time based on interval
-func credChunkFn(interval timeutils.ISODuration) func(time.Time, bool) time.Time {
-	return func(t time.Time, next bool) time.Time {
-		var result time.Time
+func credChunkFn(interval timeutils.ISODuration) func(time.Time) (time.Time, time.Time) {
+	return func(t time.Time) (time.Time, time.Time) {
+		var (
+			start time.Time
+			end   time.Time
+		)
```

Comment:
it might be slightly simpler to truncate the input date first and then simply add our ISODuration